src/main/java/com/czertainly/core/auth/oauth2/CzertainlyOAuth2FailureHandler.java

```diff
@@ -4,24 +4,31 @@
 import jakarta.servlet.http.HttpServletResponse;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.security.core.AuthenticationException;
 import org.springframework.security.web.authentication.AuthenticationFailureHandler;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
 
 import java.io.IOException;
 import java.net.URLEncoder;
 import java.nio.charset.StandardCharsets;
 
+@Component
 public class CzertainlyOAuth2FailureHandler implements AuthenticationFailureHandler {
 
     private static final Logger logger = LoggerFactory.getLogger(CzertainlyOAuth2FailureHandler.class);
 
+    @Value("${server.servlet.context-path}")
+    private String contextPath;
+
     @Override
     public void onAuthenticationFailure(HttpServletRequest request, HttpServletResponse response, AuthenticationException exception) {
         logger.debug("Error occurred when trying to authenticate using OAuth2 protocol: {}", exception.getMessage());
         try {
-            response.sendRedirect("/api/login?error=" + URLEncoder.encode(exception.getMessage(), StandardCharsets.UTF_8));
+            response.sendRedirect(contextPath + "/login?error=" + URLEncoder.encode(exception.getMessage(), StandardCharsets.UTF_8));
```

Comment:
Is this going to work when the application is running behind proxy or gateway? What about something like `ServletUriComponentsBuilder.fromCurrentContextPath().build().toUriString()`?